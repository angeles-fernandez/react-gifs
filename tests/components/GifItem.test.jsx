import { GifItem } from "../../src/components/GifItem";
import { render, screen } from '@testing-library/react';

describe('Pruebas en <GifItem />', () => {
    const title = "Saitama";
    const url = "htpps://one-punch.com/saitama.jpg"

    test('deber de dar match con el snapshot', () => {

        const { container } = render( <GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('deber de mostrar la imagen con el URL y el ALT indicado', () => {

        render(<GifItem title={title} url={url} />);
        expect(screen.getByRole('img').src).toBe(url);
        expect(screen.getByRole('img').alt).toBe(title);

    });

    test('deber de mostrar el titulo del componente', () => {

        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    });
})