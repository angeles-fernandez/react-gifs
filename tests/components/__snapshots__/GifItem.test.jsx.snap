// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> deber de dar match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="htpps://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
